src/game/client/ingameui.cpp

```diff
@@ -63,3 +1158,597 @@ bool InGameUI::Are_Selected_Objects_Controllable()
+        obj = draw->Get_Object();
+    } else {
+        obj = nullptr;
+    }
+
+    bool can_select = Can_Select_Drawable(draw, false);
+
+    if (obj == nullptr) {
+        can_select = false;
+    }
+
+    Drawable *draw2 = nullptr;
+    Object *obj2 = nullptr;
+
+    if (Get_Select_Count() == 1) {
+        draw2 = Get_All_Selected_Drawables()->front();
+
+        if (draw2 != nullptr) {
+            obj2 = draw2->Get_Object();
+        } else {
+            obj2 = nullptr;
+        }
+    }
+
+    if (m_mouseMode != MOUSEMODE_DEFAULT) {
+        if (m_mouseMode == MOUSEMODE_BUILD_PLACE) {
+            if (b) {
+                Set_Mouse_Cursor(CURSOR_ARROW);
+                return;
+            }
+
+            if (type > GameMessage::MSG_DO_ATTACK_OBJECT_AFTER_MOVING_HINT) {
+                if (type != GameMessage::MSG_ADD_WAYPOINT) {
+                    return;
+                }
+            } else {
+                if (type == GameMessage::MSG_DO_ATTACK_OBJECT_AFTER_MOVING_HINT
+                    || type == GameMessage::MSG_DO_ATTACK_OBJECT_HINT) {
+                    Set_Mouse_Cursor(CURSOR_INVALID_BUILD);
+                    return;
+                }
+
+                if (type <= GameMessage::MSG_DOCK_HINT || type > GameMessage::MSG_DO_ATTACKMOVETO_HINT) {
+                    return;
+                }
+            }
+
+            Set_Mouse_Cursor(CURSOR_BUILD);
+            return;
+        }
+
+        if (m_mouseMode != MOUSEMODE_GUI_COMMAND) {
+            return;
+        }
+
+        if (b) {
+            Set_Mouse_Cursor(CURSOR_ARROW);
+            return;
+        }
+
+        if (m_pendingGUICommand != nullptr) {
+            if (m_pendingGUICommand->Is_Context_Command() || m_pendingGUICommand->Get_Command() == GUI_COMMAND_SPECIAL_POWER
+                || m_pendingGUICommand->Get_Command() == GUI_COMMAND_SPECIAL_POWER_FROM_SHORTCUT) {
+                Utf8String cursor_name;
+
+                if (type == GameMessage::MSG_VALID_GUICOMMAND_HINT) {
+                    cursor_name = m_pendingGUICommand->Get_Cursor_Name();
+                } else {
+                    cursor_name = m_pendingGUICommand->Get_Invalid_Cursor_Name();
+                }
+
+                MouseCursor cursor = g_theMouse->Get_Cursor_Index(cursor_name);
+
+                if (cursor == CURSOR_INVALID) {
+                    Set_Mouse_Cursor(CURSOR_TARGET);
+                } else {
+                    Set_Mouse_Cursor(cursor);
+                }
+
+                Set_Radius_Cursor(m_pendingGUICommand->Get_Radius_Cursor_Type(),
+                    m_pendingGUICommand->Get_Special_Power(),
+                    m_pendingGUICommand->Get_Weapon_Slot());
+            } else if ((m_pendingGUICommand->Get_Options()
+                           & (COMMAND_OPTION_CONTEXTMODE_COMMAND | COMMAND_OPTION_NEED_TARGET_POS
+                               | COMMAND_OPTION_NEED_TARGET_ALLY_OBJECT | COMMAND_OPTION_NEED_TARGET_NEUTRAL_OBJECT

```

Comment:
Or combine last 3 to COMMAND_OPTION_NEED_TARGET_OBJECT

Reply:
fixed.